src/bridge/AdminRoomHandler.ts

```diff
@@ -420,8 +435,81 @@ export class AdminRoomHandler {
         }
     }
 
+    private async handleReconnect(req: BridgeRequest, server: IrcServer, clientList: BridgedClient[]) {
+        const client = clientList.find((c) => c.server.domain === server.domain);
+        try {
+            if (client) {
+                await client.disconnect("iwanttoreconnect", "Reconnecting", false);
+                return new MatrixAction(
+                    "notice", `Reconnecting to network...`
+                );
+            }
+            return new MatrixAction(
+                "notice", `No clients connected to this network, not reconnecting`
+            );
+        }
+        catch (err) {
+            req.log.error(err.stack);
+            return new MatrixAction(
+                "notice", `Failed to reconnect`
+            );
+        }
+    }
+
+    private async handleUsername(req: BridgeRequest, args: string[], userId: string, server: IrcServer) {
+        const domain = server.domain;
+        const store = this.ircBridge.getStore();
+        let notice;
+
+        try {
+            // Allow passwords with spaces
+            const username = args[0]?.trim();
+            if (!username || username.length === 0) {
```

Comment:
Empty strings are falsy.
```suggestion
            if (!username) {
```